sympy/polys/polymatrix.py

```diff
@@ -0,0 +1,77 @@
+from __future__ import print_function
+
+from sympy.matrices.dense import MutableDenseMatrix
+from sympy.polys.polytools import Poly
+from sympy.core.compatibility import reduce
```

Comment:
SymPy's `reduce` is, in fact, Python's reduce. Only it has been moved from builtins to functools in Python 3. It seems that its use is discouraged in favour of an explicit loop construction.